server/resourcemanager/testutil/operations.go

```diff
@@ -1,115 +1,47 @@
 package testutil
 
 import (
-	"encoding/json"
-	"io"
 	"net/http"
+	"net/http/httptest"
 	"testing"
-
-	"github.com/stretchr/testify/require"
 )
 
+type Operation string
+
+type operationTester struct {
+	buildRequest   func(*testing.T, *httptest.Server, contentTypeConverter, ResourceTypeTest) *http.Request
+	assertResponse func(*testing.T, *http.Response, contentTypeConverter, ResourceTypeTest)
+	name           Operation
+	postAssert     func(t *testing.T, ct contentTypeConverter, rt ResourceTypeTest, testServer *httptest.Server)
+}
+
 var (
 	defaultOperations = []operationTester{
-		createNoIDOperation{},
-		createSuccessOperation{},
+		createNoIDOperation,
+		createSuccessOperation,
 
-		updateNotFoundOperation{},
-		updateSuccessOperation{},
+		updateNotFoundOperation,
+		updateSuccessOperation,
 
-		getNotFoundOperation{},
-		getSuccessOperation{},
+		getNotFoundOperation,
+		getSuccessOperation,
 
-		deleteNotFoundOperation{},
-		deleteSuccessOperation{},
+		deleteNotFoundOperation,
+		deleteSuccessOperation,
 
-		listNoResultsOperation{},
-		listSuccessOperation{},
+		listNoResultsOperation,
+		listSuccessOperation,
+		listWithInvalidSortFieldOperation,
```

Comment:
you can move this to `errorOperations` since the test is actually testing how the resourcemanager itself handles an error, and is not directly related to each of the implementing resources

Reply:
Ok!